src/main/java/org/folio/fqm/service/FqlToSqlConverterService.java

```diff
@@ -68,145 +72,168 @@ public Condition getSqlCondition(String fqlCriteria, EntityType entityType) {
   /**
    * Converts the given FQL condition to the corresponding SQL query
    */
-  public Condition getSqlCondition(FqlCondition<?> fqlCondition, EntityType entityType) {
-    return sqlConverters.getOrDefault(fqlCondition.getClass(), (c, e) -> falseCondition()).apply(fqlCondition, entityType);
+  // The unchecked warning is suppressed because sqlConverters has to have a raw type, forcing its use to have an unchecked call
+  @SuppressWarnings("unchecked")
+  public static Condition getSqlCondition(FqlCondition<?> fqlCondition, EntityType entityType) {
+    final Field<Object> field = fqlCondition instanceof FieldCondition<?> fieldCondition
+      ? field(fieldCondition, entityType)
+      : null;
```

Comment:
This is one of the changes that caused this PR to be so big: getting the field object is somewhat expensive and almost every method was doing it at least once, so I changed it so that we get the field object right off the bat, then pass it through to all of the methods that need it.

Reply:
> so big
I guess I haven't given you enough of my PRs yet...

Reply:
I guess "noisy" is probably a better description :D